packages/react-native-web/src/exports/Image/__tests__/index-test.js

```diff
@@ -174,6 +171,38 @@ describe('components/Image', () => {
       expect(onLoadEndStub.mock.calls.length).toBe(2);
     });
 
+    test('is called on update if "headers" are different', () => {
+      const onLoadStartStub = jest.fn();
+      const onLoadStub = jest.fn();
+      const onLoadEndStub = jest.fn();
+      const { rerender } = render(
+        <Image
+          onLoad={onLoadStub}
+          onLoadEnd={onLoadEndStub}
+          onLoadStart={onLoadStartStub}
+          source={{
+            uri: 'https://test.com/img.jpg',
+            headers: { 'x-custom-header': 'abc123' }
+          }}
+        />
+      );
+      act(() => {
+        rerender(
+          <Image
+            onLoad={onLoadStub}
+            onLoadEnd={onLoadEndStub}
+            onLoadStart={onLoadStartStub}
+            source={{
+              uri: 'https://test.com/img.jpg',
+              headers: { 'x-custom-header': '123abc' }
+            }}
+          />
+        );
+      });
+      expect(onLoadStub.mock.calls.length).toBe(2);
+      expect(onLoadEndStub.mock.calls.length).toBe(2);
```

Comment:
Similarly to existing test, that verifies we try to reload an image when the URI changes
This test verifies that we try to reload the image if source headers change
That's why we anticipate 2 `onLoad` calls - 1 for initial load and 2 for the reload 

Because hooks and side effects are involved this test verifies we do trigger a reload only when necessary